### PR TITLE
Feat/dy

### PR DESCRIPTION
**Pull request by simlu**

### CODE REVIEW

File: src/module/dy.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+      name,
+      attributes,
+      indices,
+      DocumentClient: getService('DynamoDB.DocumentClient')
+    });
+    return ({
+      model,
+      update: () => {},
+      get: () => {},
+      genSchema: () => null // subset of cloudformation template
```
Comment (simlu) [inline src/module/dy.js:19]:
```suggestion
      genSchema: model.genSchema
```

File: src/module/dy/create-model.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    attributes,
+    table
+  });
+
+  const defined = [
+    partitionKey,
+    ...(sortKey === undefined ? [] : [sortKey])
+  ];
+
+  const schema = {
```
Comment (simlu) [inline src/module/dy/create-model.js:57]:
```suggestion
  const genSchema = () => {
```

File: test/dy-helper.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    region: 'us-west-2',
+    accessKeyId: 'XXXXXXXXXXXXXXXXXXXX',
+    secretAccessKey: 'XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX'
+  });
+  return ddb[cmd](params).promise();
+};
+
+module.exports.LocalTable = (model) => ({
+  create: async () => dynamoDB('createTable', model.schema),
+  tearDown: async () => dynamoDB('deleteTable', { TableName: model.table.name })
```
Comment (simlu) [inline test/dy-helper.js:16]:
```suggestion
  delete: async () => dynamoDB('deleteTable', { TableName: model.table.name })
```

File: test/module/dy.spec.js [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+    model = Model({
+      name: 'table-name',
+      attributes: {
+        id: { type: 'string', partitionKey: true },
+        name: { type: 'string', sortKey: true },
+        age: { type: 'number' }
+      },
+      indices: {
+        targetIndex: {
+          // todo: this should error the way that is is
```
Comment (simlu) [inline test/module/dy.spec.js:45]:
```suggestion
```

File: test/module/dy.spec.js [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+      age: 50
+    };
+  });
+  afterEach(async () => {
+    await localTable.tearDown();
+  });
+
+  it('Testing basic logic', () => {
+    expect(Object.keys(model)).to.deep.equal([
+      'model',
```
Comment (simlu) [inline test/module/dy.spec.js:68]:
```suggestion
      '_model',
```

Comment (simlu) [reply test/module/dy.spec.js:68]: Remove if we can, check

File: test/module/dy.spec.js [hunk 6]
```diff
@@ -0,0 +1,223 @@
...
+      page: {
+        next: null,
+        index: { current: 1 },
+        size: 20
+      }
+    });
+  });
+
+  it('Testing query with limit', async () => {
+    expect(await model.upsert(item)).to.deep.equal({ created: true });
```
Comment (simlu) [inline test/module/dy.spec.js:158]: should probably insert two

File: test/module/dy.spec.js [hunk 7]
```diff
@@ -0,0 +1,223 @@
...
+      page: {
+        next: null,
+        index: { current: 1 },
+        size: 20
+      }
+    });
+  });
+
+  it('Testing query with cursor', async () => {
+    expect(await model.upsert(item)).to.deep.equal({ created: true });
```
Comment (simlu) [inline test/module/dy.spec.js:204]: add two and use first cursor. Don't magically use cursor, genereate cursor and then use it

**Comment by simlu** [comment]
Do readme and backmerge

**Review by krackerman — APPROVED** [review]

**Pull request merged**